### ISSUE TITLE
Product Engineering Review : Parkinsons Disease Prediction

### ISSUE THREAD

**Issue by shilpa-nalkande** [Teradata/jupyter-demos#536]

**Comment by shilpa-nalkande** [comment]
Reviewer 1 comments:
Dec 1, 2023
under section 3, second cell executes SQL call with old sqlalchemy engine object. This should be changed to execute_sql
Under section 4, they used a proc to load the data. We can use fastload or copy_to_sql.
section 6 mentions some warnings. We/teradataml should look at those and identify the cause for those.
section 6 has model generation using scikit-learn. We should use open source ML for it or Open AF.
section 6 uses bar plot. But, with matplotlib. Use teradataml inhouse plot.
even in section 7, data is being pulled to local using to_pandas. This should be changed.
section 8 uses SQL. This should be changed to teradataml function.
section 9 uses matplotlib plot. This should be changed.
Reviewer 2 comments:
Data splitting into test and train should use teradataml sample function.
@Pankaj Vinod Purandare
Comments:
Section 6: (Enhancement)
Latest version of teradataml allows user to execute SQLE analytic function using the model itself. For example, you are running ScaleFit and the ScaleTransform. With new support you can do following:
sf_trns = ScaleFit()
sf_trns.transform() – No need to pass the object argument to it.
Section 6:
Use of “LogisticRegression”, why don’t we use SQLE or VAL function to
take care of this? This does not demonstrates the ClearScape analytics capabilities.
Section 6:
Use of “train_test_split” from SKL, why don’t we DataFrame.sample() or TrainTestSplit() functions?
In section “Train and Test Split using Sample”
Noticed that we are using DataFrame.sample() function to get the sample and we are getting two dataframes.
Then those DataFrames are being persisted using the copy_to_sql(), which is not necessary. These teradataml DataFrames can be used as is for further operations as is.
pandas read query is being run to get the pandas dataframe. This is unnecessary step. We could have printed the teradataml dataframe as is.
Converting teradataml DataFrame to pandas DataFrame using to_pandas() is recommended it only when teradataml/Teradata doesn’t have an option to take care of the operations.
Avoid “Pd_” prefix for teradataml DataFrames names or Teradata table names, as that denotes the Pandas.
Multiple places copy_to_sql() or to_sql() is used. These should be used only when user wants to persist the DataFrames. 
This step is recommended to be performed right before the step when user wants to log off the session. As all dataframes are garbage collected, so that they can resume their session next day from the step they left off.
copy_to_sql() can be used to persist the teradataml DataFrame or main purpose is to bring outside data from Pandas DataFrame to Vantage.
to_sql() can be used to persist the teradataml DataFrame.
Section 7: In Decision Tree Model Section:
Header says Decision Tree, but we are using DecisionForest.
Section 7.1:
When a call made to `DecisionForest` function, we could have passed the teradataml DataFrame created after the sample operation. No need to persist the DF and pass it.
Section 7.2:
With latest teradataml release, you can use the `DecisionForest` model object to execute the predict and evaluate. No need to call `
DecisionForestPredict` and `Evaluator` function. On can do following:
obj = DecisionForest()
obj.predict() à Pass all other required args, except object.
obj.evaluate() à Pass all other required args, except object.
Section 7.3
What is the purpose of running copy_to_sql?
Instead of the SQL execution of ` TD_ClassificationEvaluator`, we could have used teradataml function calls.
Using pd.read_sql() undermines the Python capabilities provided by Teradata. We could have simply used the teradataml DataFrame on the table created.
Section 8:
Comments given for section 7.2 and 7.3 are applicable for this section too.
Section 9:
To compare output of GLM and Decision Forest metric, we could operated on the output results which are teradataml DataFrames. We could have run the Union() function from teradataml to get the desired output.
SQL query is not needed at all.
Teradataml provided a functionality to pivot the results. We could have
used the same instead of pandas.
To plot a BAR plot for visualization, teradataml offer In-DB plotting, no data transfer. Use DataFrame.plot() functionality for the same.
